### ISSUE TITLE
get() slow for single pixel, add fget()?

### ISSUE THREAD

**Issue by smoothmango** [processing/p5.js#2065]
When extracting single pixels from a PGraphics I noticed that `get(x,y)` is extremely slow. For instance, when extracting 220 pixel color values from a 1920x1080 pixel sketch, using `get(x,y)` took 4800ms. I overloaded PGraphics with my own "fast get" function `fget(x,y)` which uses offset in the `pixels[]` array as recommended [here](https://p5js.org/reference/#/p5/get) and was able to do the same work in 0.325ms. That's 15,000x faster!

Note the reason for this massive speed difference is that p5js' calls `loadPixels()` inside each `get(x,y)`, which in my case takes 25ms.

Given the 15,000x speed difference, I propose that p5js add an `fget(x,y)` "fast get" or maybe `pget(x,y)` "pixel get" which implements getting a single pixel color via offset in the pixels[] array as opposed to the fancery that goes on in `get(x,y)`. Use of this function would require the user call `loadPixels()` first, and would therefore be a little beginner unfriendly. 

TL;DR => let's add a "fast get" function to extract single pixels from a PGraphics, it will be 15,000x faster than `get(x,y)`

**Comment by sixhat** [comment]
What about a version of `get` that checks if `loadPixels` has been called in the current frame already? Is that possible? My reasoning is that `get` is slow when typically one is trying to load a bunch of pixels from the same image and therefore `loadPixels` only needs to be called once. Maybe some tests can clear is this is possible, but there might be other use cases where `get` needs to `loadPixels` always.

**Comment by lmccart** [comment]
hi @smoothmango I see your point. and we thought a lot about this when deciding whether to call loadPixels() inside the function or require the user to do it. we eventually settled on this method being rather slow but easy to use, and then having the `pixels[]` array to give the option for fast access.

it wouldn't be too hard to write your own fget() function that grabs things out of this array. this would be even faster than calling get() since there is still other checking and logic going on in this function that's unnecessary for the case where you just need one value from the pixels array. I think between these two options we're covered, and would prefer to limit the API here.

**Comment by smoothmango** [comment]
I understand your position, perhaps it would be worth adding a more stern warning about the performance implications of get() in the documentation? It's a huge performance killer, and seems to me to be more painful in js than it was in java Processing. It kills nearly 2 frames of animation at 60fps to check even a single pixel at HD 1920x1080 resolution!

**Comment by MichaelPaulukonis** [comment]
The difference compared to processing.js is staggering.

**Comment by Spongman** [comment]
@michaelpaulukonis can you possibly post a sample that demonstrates the performance issue you’re seeing? It’s not surprising that Java is quicker in general, but there have been changes in this area recently to address this and it _should_ be improved, but there may well be more room for improvement...

**Comment by rodigu** [comment]
`fget` would be a great feature!
